doc/apiary/v2/fiware-ngsiv2-reference.apib

```diff
@@ -713,8 +713,65 @@ so notification receiver can know the format used without needing processing not
 
 ### Notification templates
 
-Based on the "template" field defined as part of the subscription and in a macro replacement language
-that could be used in "payload", "url", "headers", "method" and "qs". Details TBD.
+At notification time, NGSIv2 implementations will send a notification based on the `notification.httpExtended`
+field of the associated subscription. These fields can be interpreted literally or use a
+templating mechanism, based in macros substitution, that is described in the present subsection.
+
+Only the following fields inside `notification.httpExtended` can be templatized:
+
+* `url`
+* `headers` (both header names and values)
+* `qs` (both in parameters names and values)
+* `payload`
+
+It is worth mentioning that `method` is not a template and it is always used literally.
+
+Macro substitution is based on the syntax `${..}`. In particular:
+
+* `${id}` is replaced by entity id
+* `${type}` is replaced by entity type
+* Any other `${token}` is replaced by the value of attribute which name is `token` or with
+  an empty string if the attribute is not included in the notification. If the value is
+  not a string (i.e. it is a number, a bool, a JSON object or array) it is casted to a string.
```

Comment:
if the payload is actually JSON, then it is still not clear to me how this would work ... I think it needs to be clarified